### PR TITLE
config: move ConfigClient out of PdClient

### PR DESCRIPTION
**Pull request by NingLin-P**
###  What have you changed?

This pr  move `ConfigClient` trait definition out of `PdClient`. This pr also write the changed `raftstore` config into metrics.

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)
- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?

- Manual test (`make clippy`)

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
No

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -1919,7 +1919,7 @@ impl ConfigHandler {
     /// version and config
     pub fn create(
         id: String,
-        pd_client: Arc<impl PdClient>,
+        pd_client: Arc<impl ConfigClient>,
```
Comment (overvenus) [inline src/config.rs]: Rename it to `cfg_client` would be better.

File: src/config.rs [hunk 2]
```diff
@@ -1947,7 +1947,7 @@ impl ConfigHandler {
 
     /// Update the local config if remote config had been changed,
     /// rollback the remote config if the change are invalid.
-    pub fn refresh_config(&mut self, pd_client: Arc<impl PdClient>) -> CfgResult<()> {
+    pub fn refresh_config(&mut self, pd_client: Arc<impl ConfigClient>) -> CfgResult<()> {
```
Comment (overvenus) [inline src/config.rs]: Rename it to `cfg_client` would be better.

File: src/config.rs [hunk 3]
```diff
@@ -1989,7 +1989,7 @@ impl ConfigHandler {
         &mut self,
         version: configpb::Version,
         entries: Vec<configpb::ConfigEntry>,
-        pd_client: Arc<impl PdClient>,
+        pd_client: Arc<impl ConfigClient>,
```
Comment (overvenus) [inline src/config.rs]: Rename it to `cfg_client` would be better.

**Comment by overvenus** [comment]
LGTM, please resolve conflicts.

File: src/raftstore/store/fsm/apply.rs [hunk 4]
```diff
@@ -2748,7 +2748,6 @@ impl PollHandler<ApplyFsm, ControlFsm> for ApplyPoller {
     fn begin(&mut self, _batch_size: usize) {
         if let Some(incoming) = self.cfg_tracker.any_new() {
             match Ord::cmp(&incoming.messages_per_tick, &self.messages_per_tick) {
-                CmpOrdering::Equal => {}
```
Comment (rleungx) [inline src/raftstore/store/fsm/apply.rs]: Why do we make this change?

Comment (NingLin-P) [reply src/raftstore/store/fsm/apply.rs]: I think this may look better...😓

**Pull request merged**